File: Beispiele/Java2Kotlin/src/main/kotlin/javakotlinworkshop/java/Conditions.java
package javakotlinworkshop.java;

public class Conditions {

    public static void main(String[] args) {
        int num = (int) (Math.random() * 4f);
        System.out.println(String.format("%d ist %s",
                num,
                isEven(num) ? "gerade" : "ungerade"));

        switch (num) {
            case 0:
                System.out.println("0");
                break;
            case 2:
                System.out.println("2");
                break;
            default:
                System.out.println("1 oder 3");
        }

        Person person = new Person();
        // the conditional operator
        String s = person.name != null ? person.name : "???";
        System.out.println(s);
    }

    private static boolean isEven(int num) {
        return (num % 2) != 1;
    }

    static class Person {
        String name;

        Person() {
            this(null);
        }

        Person(String name) {
            this.name = name;
        }
    }
}

File: Beispiele/Java2Kotlin/src/main/kotlin/javakotlinworkshop/kotlin/Conditions.kt
package javakotlinworkshop.kotlin

fun main(args: Array<String>) {
    val num = (Math.random() * 4).toInt()
    println("$num ist ${if (isEven(num)) "gerade" else "ungerade"}")

    // branch conditions may also be combined with a comma
    // ranges can be checked like this: in 1..10
    when (num) {
        0 -> println("0")
        2 -> println("2")
        else -> println("1 oder 3")
    }

    val person = object {
        val name = null
    }
    val s = person.name ?: "???"
    println(s)
}

fun isEven(num: Int): Boolean {
    return (num % 2) != 1
}